netket/sampler/metropolis.py

```diff
@@ -136,9 +136,12 @@ class MetropolisSamplerState(SamplerState):
     """State of the random number generator (key, in jax terms)."""
     rule_state: Optional[Any]
     """Optional state of a transition rule."""
-    n_steps_proc: int = 0
+
+    # those are initialised to 0. We want to initialise them to zero arrays because they'll
+    # be used inside of jax loops that require type invariance.
```

Comment:
```suggestion
    # those are initialised to 0. We want to initialise them to zero arrays because they can
    # be passed to jax jitted functions that require type invariance to avoid recompilation
```
(this should ensure they are not converted to weakly typed arrays)

I'll try to test if it actually works as intended later